src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -983,20 +989,16 @@ public void TrimExcess(int capacity)
             _version++;
             Initialize(newSize);
 
-            Debug.Assert(oldEntries is not null);
```

Comment:
Similar comment - unclear why this is needed for Roslyn build.

Reply:
➡️ `oldEntries` is a non-nullable value type in `SegmentedDictionary<TKey, TValue>` (it changed from `Entry[]` in .NET Runtime to `SegmentedArray<Entry>` here).